### DISCUSSION TITLE
Creating curved subdomains by use of cylinders?

### DISCUSSION THREAD

**Discussion by mustafasubasic** [neperfepx/neper#653] [Q&A] [answered]
Dear all

I am trying to create a curved layered domain by use of two cylinders. I want the cylinders to be equi-axed with the same height. The only difference should be the diameters and the number of grains. I want the smaller cylinder to have 100 grains while the bigger cylinder should be blank ("1 grain"). Is this possible in Neper?

Thank you!

Best regards
Mustafa Subasic

**Comment by rquey** [comment, marked as answer]
I don't really see what you mean. Can you make a sketch?

**Reply by mustafasubasic** [reply]
See image below: a smaller cylinder with grains inside a bigger blank cylinder. From this I will then create my model by performing some transformation commands but that is no problem as long as I can get the two cylindrical domains below. Thank you very much!

![tess](https://user-images.githubusercontent.com/126886091/222802643-881bdb09-beec-41d3-96eb-a72888af35e2.PNG)

**Reply by rquey** [reply]
Thanks. Neper doesn't allow this. Neper allows to generate the tessellation at the center but not the part surrounding it. At least this is true if you want a perfect / cylindrical interface between the two.

So; you'll need to do some sort of post-processing.

**Reply by mustafasubasic** [reply]
Ok. Thank you very much for the quick answers! I appreciate it.

**Comment by charanjo** [comment]
HI, 
I am trying to create a voronoi domain for hour glass shaped specimen (refer image) with varying grain size? I need one grain size at the center of the specimen and different grain size near the edges? Is it possible to generate such domain for this shape?  
![Picture1](https://github.com/user-attachments/assets/85254e9e-3e55-4e56-ae6d-279a00054ca7)
